### DISCUSSION TITLE
Get-DbaUserPermission option to create the STIG schema in another database

### DISCUSSION THREAD

**Discussion by griffitmatt** [dataplat/dbatools#9414] [Ideas]
Hi all, 

Would it be possible to have an option to supply another database name instead of using the tempdb?  

For accounts with limited permissions (locked down environments) on server and databases this would allow the option to have a specific user database setup with the necessary permissions instead of applying them to the tempdb. IE permissions to create the STIG schema and objects.

Thanks

Matt

**Comment by niphlod** [comment]
how can it be so locked down tempdb isn't accessible ?

**Reply by griffitmatt** [reply]
> how can it be so locked down tempdb isn't accessible ?

The issue is creating objects specifically in the tempdb (STIG) . Creating temporary objects are allowed for any user in the tempdb. Eg #table but is not the case for permanent objects.  The error being returned is Failed to create or use STIG schema on ########## | User does not have permission to perform this action. CREATE SCHEMA failed due to previous errors.
